verifier/src/report_body.rs

```diff
@@ -315,6 +316,40 @@ impl<T: Accessor<MrSigner>> Verifier<T> for MrSignerVerifier {
     }
 }
 
+/// Verify the [`ReportData`] is as expected.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ReportDataVerifier {
+    expected_report_data: ReportData,
+}
+
+impl ReportDataVerifier {
+    /// Create a new [`ReportDataVerifier`] instance.
+    ///
+    /// # Arguments:
+    /// * `expected_report_data` - The expected report data.
+    pub fn new(expected_report_data: ReportData) -> Self {
+        Self {
+            expected_report_data,
+        }
+    }
+}
+
+impl<T: Accessor<ReportData>> Verifier<T> for ReportDataVerifier {
+    type Error = VerificationError;
+
+    fn verify(&self, evidence: &T) -> CtOption<Self::Error> {
+        let expected = self.expected_report_data.clone();
+        let actual = evidence.get();
+
+        // TODO - This should be a constant time comparison.
+        let is_some = if expected == actual { 0 } else { 1 };
```

Comment:
It might make sense to make this constant-time before merging the PR, although I don't know what would be involved in implementing a CT comparison on this type.

Reply:
Good call out!

I created #27 to track this, currently it's blocked on https://github.com/mobilecoinfoundation/sgx/issues/280